[RFC] Load-adjust timer tests

mhinz opened the pull request:
- switch to Xcode 10.1
- calculate load and adjust timer tests
- fix `Test_help_tagjump()`

mhinz committed:
ci: switch to Xcode 10.1 / macOS 10.13

justinmk commented:
can this be closed?

justinmk commented:
If someone wants to do that, sure. It's not a one-time task: more tests will fail over time, and those need to be monitored, and later fixed.

mhinz commented:
Then I try looking into the issues of this PR (after I had a look how you handled similar issues earlier).

justinmk commented on src/nvim/testdir/test_lambda.vim:
```diff
@@ -31,15 +31,13 @@ function! Test_lambda_with_timer()
   call s:Foo()
-  sleep 210ms
-  " do not collect lambda
-  call test_garbagecollect_now()
```
why remove this? it is part of the upstream test.

mhinz replied:
Happened when I played around and removed all potential error sources. Will add back.

mhinz committed:
tests: load-adjust timer tests (oldtest)

mhinz committed:
tests: load-adjust timer tests (functionaltest)

mhinz commented:
I think this PR is ready now.  💪 

(The failing macOS jobs are due to the unrelated `Test_help_tagjump()` old test.)

mhinz committed:
tests: fix Test_help_tagjump()

The Vim version of Test_help_tagjump() tests for `:help sm?le` here. That
command got removed from Nvim, so the test was changed to check against `:help
sp?it` instead.

The new test already handled the case that on Win `:h split` would jump to the
entry for split() and on all other systems to the entry for :split.

Then this commit happened:

   bb3aa824b lua/stdlib: vim.inspect, string functions

Since then `:h split` would jump to split() for macOS as well! I'm not sure why.

Anyway, instead of adding another check for has('mac'), we change the test once
more to be more akin to the original test. Instead of testing for :smile, which
is exclusive to Vim, we check against :checkhealth, which is exclusive to Nvim.

justinmk commented on src/nvim/testdir/test_timers.vim:
```diff
@@ -14,13 +15,17 @@ func MyHandlerWithLists(lists, timer)
   call assert_equal(1, g:val)
   if has('reltime')
-    call assert_inrange(40, 120, slept)
+    call s:assert_inrange(40, 120, slept)
```
would be clearer if LoadAdjust() was explicitly called here

mhinz replied:
Yup, but you merged it already. :P I guess we just silently do it the next time that file is changed in any way.
